sha1_mb/sha1_mb_x16_avx512.asm

```diff
@@ -515,27 +515,12 @@ lastLoop:
 
 section .data
 align 64
-K00_19:			dq 0x5A8279995A827999, 0x5A8279995A827999
-			dq 0x5A8279995A827999, 0x5A8279995A827999
-			dq 0x5A8279995A827999, 0x5A8279995A827999
-			dq 0x5A8279995A827999, 0x5A8279995A827999
-K20_39:                 dq 0x6ED9EBA16ED9EBA1, 0x6ED9EBA16ED9EBA1
-			dq 0x6ED9EBA16ED9EBA1, 0x6ED9EBA16ED9EBA1
-			dq 0x6ED9EBA16ED9EBA1, 0x6ED9EBA16ED9EBA1
-			dq 0x6ED9EBA16ED9EBA1, 0x6ED9EBA16ED9EBA1
-K40_59:                 dq 0x8F1BBCDC8F1BBCDC, 0x8F1BBCDC8F1BBCDC
-			dq 0x8F1BBCDC8F1BBCDC, 0x8F1BBCDC8F1BBCDC
-			dq 0x8F1BBCDC8F1BBCDC, 0x8F1BBCDC8F1BBCDC
-			dq 0x8F1BBCDC8F1BBCDC, 0x8F1BBCDC8F1BBCDC
-K60_79:                 dq 0xCA62C1D6CA62C1D6, 0xCA62C1D6CA62C1D6
-			dq 0xCA62C1D6CA62C1D6, 0xCA62C1D6CA62C1D6
-			dq 0xCA62C1D6CA62C1D6, 0xCA62C1D6CA62C1D6
-			dq 0xCA62C1D6CA62C1D6, 0xCA62C1D6CA62C1D6
+K00_19:			dd 0x5A827999
+K20_39:                 dd 0x6ED9EBA1
+K40_59:                 dd 0x8F1BBCDC
+K60_79:                 dd 0xCA62C1D6
 
 PSHUFFLE_BYTE_FLIP_MASK: dq 0x0405060700010203, 0x0c0d0e0f08090a0b
```

Comment:
This needs to be aligned to 64 bytes (same for MASK2 below)

Reply:
I didn't align PSHUFFLE_BYTE_FLIP_MASK because the 4 32bit constant above it are aligned to 64 bytes, so +4*32 bit makes it aligned to 16 bytes which is the natural size for the 128bit load. You're right about the transpose masks, better to keep them aligned to avoid a split-load.

Reply:
> I didn't align PSHUFFLE_BYTE_FLIP_MASK because the 4 32bit constant above it are aligned to 64 bytes, so +4*32 bit makes it aligned to 16 bytes which is the natural size for the 128bit load. You're right about the transpose masks, better to keep them aligned to avoid a split-load.

The problem is not the split-load. I got a seg fault because there are aligned load instructions that need this data to be 64-byte aligned.

Reply:
Oh, that's interesting, i didn't get any SEGFAULT when doing `make test'. Anyway now they should be aligned, is it working ok for you?

Reply:
Now yes :) Will merge this soon, thanks!

Reply:
BTW, if you want i can also make a quick separate pull requests to use VPTERNLOG in SM3 for the boolean functions. It's such a nice instruction that it's a shame not to use it wherever possible ;)

Reply:
If you promise it's a short one that you can send in the next few hours :P

Reply:
I'll give it a try :P